File: multilanguage/demo_grpc/java/sapphire-java-runtime/src/main/java/hankstodo/app/TodoList.java
package hankstodo.app;

import java.util.ArrayList;

public class TodoList {
	ArrayList<Object> toDos = new ArrayList<Object>();
	String name = "Hanks todo";

	public TodoList(String name) {
		toDos = new ArrayList<Object>();
		this.name = name;
	}

    public String getName() {
        return name;
    }

	public String addToDo(String todo) {
		System.out.println("Inside todo: " + todo);
		toDos.add(todo);
		return "OK!";
	}

	public void completeToDo(String todo) {
	}

	public ArrayList<Object> getHighPriority() {
		return new ArrayList<Object>();
	}
}

File: sapphire/examples/hanksTodoRuby/src/main/ruby/sapphire/appexamples/hankstodo/todo_list.rb
#!/usr/bin/ruby

class TodoList
    def initialize(name)
        @to_dos = Array.new
        @name = name
    end

    def addTodo(todo)
        @to_dos.push(todo)
        puts "add #{todo} success!!!"
        return "OK!"
    end

    def completeTodo(todo)
        @to_dos.delete(todo)
        puts "#{todo} task removed"
        return "OK!"
    end

    def getTodos
        return @to_dos
    end

    def getName
        @name
    end
end
